public/js/reducers/sources.js

```diff
@@ -100,10 +96,19 @@ function update(state = State(), action: Action) : Record<SourcesState> {
       }
 
       return _updateText(state, action, [action.originalSource]);
+
     case "NAVIGATE":
-      // Reset the entire state to just the initial state, a blank state
-      // if you will.
-      return State();
+      if (!action.data) {
+        return state;
+      }
+      if (!state.url) {
+        return State()
+          .set("url", action.data.url);
+      }
+      if (state.url === action.data.url) {
+        return state;
+      }
+      return State().set("url", action.data.url);
```

Comment:
this code can be simplified by following up on your previous comments.

`targetUrl` could probably make sense for tabs and node also. Although, I haven't looked into node yet.

I am using `url` to check if the new target url is same as the old one, and based on that returning `state` or new `State()`

I'll wait for jlongster's thoughts on these changes.